src/main/java/seedu/address/model/assignment/UniqueAssignmentList.java

```diff
@@ -42,6 +44,28 @@ public boolean contains(Assignment toCheck) {
         return internalList.stream().anyMatch(toCheck::isSameAssignment);
     }
 
+    /**
+     * Returns true if the list contains an equivalent assignment as the given
+     * argument.
+     */
+    public boolean contains(AssignmentId toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(assignment -> assignment.getAssignmentId().equals(toCheck));
+    }
+
+    /**
+     * Returns true if the list contains an equivalent assignment as the given
+     * argument.
+     */
+    public boolean contains(ProjectId projectToCheck, EmployeeId employeeToCheck) {
+        requireNonNull(projectToCheck);
+        requireAllNonNull(employeeToCheck);
+        return internalList.stream().anyMatch(assignment ->
+                assignment.getPerson().getEmployeeId().equals(employeeToCheck)
+                        && assignment.getProject().getId().equals(projectToCheck));
+    }
```

Comment:
Would using the overloaded `Assignment#isSameAssignment()` method be applicable here? I think this would adhere to the DRY principal better.

Reply:
Sure! I changed it in latest commit